libs/js/Date.jsx

```diff
@@ -0,0 +1,53 @@
+import React, {Component} from 'react';
+import {
+  FormGroup,
+  FormControl,
+  HelpBlock
+} from 'react-bootstrap';
+export default class Text extends Component {
+  constructor(props){
+    super(props);
+    this.state = {
+      value : props.value,
+    };
+  }
+  getValue = () =>{
+    return this.state.value;
+  }
+  setValue = (e) =>{
+    this.setState({ value: e.target.value });
+    this.props.setValueToAnchor(e.target.value, e);
+  }
+  onBlur(e){
+    if( !this.props.showButtons ){
+      this.setValue(e);
+      this.props.onSubmit();
+    }
+  }
+  handleKeyDown(target) {
+    if(target.keyCode == 13){
+      this.setValue(target);
+      this.props.onSubmit();
+    }else if(target.keyCode == 27){
+      this.props.setEditable(false)
+    }
+  }
+  render(){
+    return (
+      <FormGroup controlId="formBasicText" validationState={this.props.validation.type} key={"FormGroup"+this.props.name}>
+        <FormControl
+          key={"form-control"+this.props.name}
+          type="date"
```

Comment:
I am surprised that react bootstrap _formControl_ has _date_  type because there is no such thing exist in the documentation. Can you please provide some brief information on this?

Reply:
Got it. Its using default browser one.

Reply:
Yeah, I usually prefer the browser default so that we dont have to worry about its styling; however, different browsers shows it differently, so can create branding issues if someone is too too concerned with things like that :)